src/Tribe/API.php

```diff
@@ -602,5 +642,222 @@ public static function get_and_flatten_event_meta( $event_id ) {
+
+			return $new_value;
+		}
+
+		/**
+		 * Sanitize a string to be used as an event hour post meta value: 1-12 if `Event{Start|End}Meridian`
+		 * is also passed, else 0-23.
+		 *
+		 * Use this prior to sending value to the database.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $value       The post meta value to be checked, such as '07'.
+		 * @param bool   $twelve_hour If false, allow 0-23. If true, only allow 1-12.
+		 *
+		 * @return string|WP_Error
+		 */
+		public static function sanitize_hour_meta_value( $value, $twelve_hour = false ) {
+			$value = (string) $value;
+
+			if ( ! is_numeric( $value ) ) {
+				$error_message = sprintf(
+					esc_html__( 'An event having a post meta value of `%s` hour (am/pm) is not valid. Make sure it is from 1 to 12.', 'the-events-calendar' ),
+					$value
+				);
+
+				return new WP_Error( 'non-numeric-tribe-events-hour-meta-value', $error_message );
+			}
+
+			$new_value = absint( $value );
+
+			if ( $twelve_hour ) {
+				if (
+					1 > $new_value
+					|| 12 < $new_value
+				) {
+					$error_message = sprintf(
+						esc_html__( 'An event having a post meta value of `%s` hour (12-hour) is not valid. Make sure it is from 1 to 12.', 'the-events-calendar' ),
+						$value
+					);
+
+					return new WP_Error( 'invalid-tribe-events-12-hour-meta-value', $error_message );
+				}
+			} else {
+				if ( 23 < $new_value ) {
+					$error_message = sprintf(
+						esc_html__( 'An event having a post meta value of `%s` hour (24-hour) is not valid. Make sure it is from 0 to 23.', 'the-events-calendar' ),
+						$value
+					);
+
+					return new WP_Error( 'invalid-tribe-events-24-hour-meta-value', $error_message );
+				}
+			}
+
+			$new_value = str_pad( $new_value, 2, '0', STR_PAD_LEFT );
+
+			return $new_value;
+		}
+
+		/**
+		 * Sanitize a string to be used as an event minute post meta value: 0-59.
+		 *
+		 * Use this prior to sending value to the database.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $value The post meta value to be checked, such as '30'.
+		 *
+		 * @return string|WP_Error
+		 */
+		public static function sanitize_minute_meta_value( $value ) {
+			$value = (string) $value;
+
+			if ( '' === trim( $value ) ) {
+				return trim( $value );
+			}
+
+			if ( ! is_numeric( $value ) ) {
+				$error_message = sprintf(
+					esc_html__( 'An event having a post meta value of `%s` minutes is not valid. Make sure it is from 0 to 59.', 'the-events-calendar' ),
+					$value
+				);
+
+				return new WP_Error( 'non-numeric-tribe-events-minutes-meta-value', $error_message );
+			}
+
+			$new_value = absint( $value );
+
+			if ( 59 < $new_value ) {
+				$error_message = sprintf(
+					esc_html__( 'An event having a post meta value of `%s` minutes is not valid. Make sure it is from 0 to 59.', 'the-events-calendar' ),
+					$value
+				);
+
+				return new WP_Error( 'invalid-tribe-events-minutes-meta-value', $error_message );
+			}
+
+			$new_value = str_pad( $new_value, 2, '0', STR_PAD_LEFT );
```

Comment:
To avoid resign this to a new variable you can return the result from `str_pad` instead.

```php
return str_pad( $new_value, 2, '0', STR_PAD_LEFT );
```

Reply:
`$new_value` was declared earlier. just changing its value here from integer to 2-digit, zero-left-padded string version of that integer. So OK?

Reply:
Same as the method before this one.

Reply:
Yeah as `str_pad` returns a string value and the operation is executed before is returned so what happens here is: 

1. First executes `str_pad`
2. A string is returned from `str_pad`
3. Value from previous string is returned from this method.

As you don't need to assign back the value again to the same variable before returning.

Reply:
it's clear to me now looking a 2nd time. I thought you meant quite a bit more lines to change.